profiler/src/main/java/com/splunk/opentelemetry/profiler/JfrActivator.java

```diff
@@ -83,35 +81,42 @@ private boolean notClearForTakeoff(ConfigProperties config) {
       return true;
     }
 
-    Path outputDir = Paths.get(config.getString(CONFIG_KEY_PROFILER_DIRECTORY));
+    return false;
+  }
+
+  private boolean checkOutputDir(Path outputDir) {
     if (!Files.exists(outputDir)) {
       // Try creating the directory for the user...
       try {
         Files.createDirectories(outputDir);
       } catch (IOException e) {
-        return outdirWarn(outputDir, "does not exist and could not be created");
+        outdirWarn(outputDir, "does not exist and could not be created");
+        return false;
       }
     }
     if (!Files.isDirectory(outputDir)) {
-      return outdirWarn(outputDir, "exists but is not a directory");
+      outdirWarn(outputDir, "exists but is not a directory");
+      return false;
     }
 
     if (!Files.isWritable(outputDir)) {
-      return outdirWarn(outputDir, "exists but is not writable.");
+      outdirWarn(outputDir, "exists but is not writable");
+      return false;
```

Comment:
The return value is duplicated here across these cases, where it used to be encapsulated in the method. I guess you just didn't like that or thought it was hard to read?

Reply:
I didn't like it.